url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Console.java

```diff
@@ -0,0 +1,623 @@
+        examples.add("/takagi-sugeno/matlab/invkine1");
+        examples.add("/takagi-sugeno/matlab/invkine2");
+        examples.add("/takagi-sugeno/matlab/juggler");
+        examples.add("/takagi-sugeno/matlab/membrn1");
+        examples.add("/takagi-sugeno/matlab/membrn2");
+        examples.add("/takagi-sugeno/matlab/slbb");
+        examples.add("/takagi-sugeno/matlab/slcp");
+        examples.add("/takagi-sugeno/matlab/slcp1");
+        examples.add("/takagi-sugeno/matlab/slcpp1");
+        examples.add("/takagi-sugeno/matlab/sltbu_fl");
+        examples.add("/takagi-sugeno/matlab/sugeno1");
+        examples.add("/takagi-sugeno/matlab/tanksg");
+        examples.add("/takagi-sugeno/matlab/tippersg");
+        examples.add("/takagi-sugeno/octave/cubic_approximator");
+        examples.add("/takagi-sugeno/octave/heart_disease_risk");
+        examples.add("/takagi-sugeno/octave/linear_tip_calculator");
+        examples.add("/takagi-sugeno/octave/sugeno_tip_calculator");
+        examples.add("/tsukamoto/tsukamoto");
+
+        Importer importer;
+        if ("fll".equals(from)) {
+            importer = new FllImporter();
+        } else if ("fis".equals(from)) {
+            importer = new FisImporter();
+        } else if ("fcl".equals(from)) {
+            importer = new FclImporter();
+        } else {
+            throw new RuntimeException("[examples error] unrecognized format "
+                    + "<" + from + "> to import");
+        }
+
+        Exporter exporter;
+        if ("fll".equals(to)) {
+            exporter = new FllExporter();
+        } else if ("fld".equals(to)) {
+            exporter = new FldExporter(" ");
+        } else if ("fcl".equals(to)) {
+            exporter = new FclExporter();
+        } else if ("fis".equals(to)) {
+            exporter = new FisExporter();
+        } else if ("cpp".equals(to)) {
+            exporter = new CppExporter();
+        } else if ("java".equals(to)) {
+            exporter = new JavaExporter();
+        } else {
+            throw new RuntimeException("[examples error] unrecognized format "
+                    + "<" + from + "> to export");
+        }
+
+        List<Op.Pair<Exporter, Importer>> tests = new ArrayList<>();
+        tests.add(new Op.Pair<Exporter, Importer>(new FllExporter(), new FllImporter()));
+        tests.add(new Op.Pair<Exporter, Importer>(new FclExporter(), new FclImporter()));
+        tests.add(new Op.Pair<Exporter, Importer>(new FisExporter(), new FisImporter()));
+
+        StringBuilder errors = new StringBuilder();
+        for (int i = 0; i < examples.size(); ++i) {
+            System.out.println("Processing " + (i + 1) + "/" + examples.size() + ": " + examples.get(i));
+            try {
+                StringBuilder text = new StringBuilder();
+                String input = sourceBase + examples.get(i) + "." + from;
+                BufferedReader source = new BufferedReader(new FileReader(input));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *source*. The resource is referenced at lines 524, 526. The resource is closed at line 529. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *source* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).